### PR TITLE
Suppress reporting of tables when reading emails in Outlook using UIA

### LINKED ISSUE #11430: Suppress layout tables when reading messages in Outlook with UIA for Word enabled
### Is your feature request related to a problem? Please describe.

When using the "Use UI Automation for Word controls" enhanced setting, reading messages in Outlook exposes way more information, and is more performant, than if this setting is off. However, in newsletters, layout tables are often used to place images or headers, or add other eye candy. With UIA for Word enabled, each and every table, nested and otherwise, is reported. This makes reading newsletters a real pain.

### Describe the solution you'd like

NVDA should apply some heuristics when reading messages in Outlook to suppress tables that are clearly for layout. It already has some heuristics to determine layout tables, and if needed, could also take some inspiration for how Firefox determines if a table is a layout table. Especially those tables that only have one column and/or row are clearly not data tables and could be suppressed safely.

### Describe alternatives you've considered

There aren't really any, since turning off UIA for Word makes the reading experience in Outlook far worse because of performance. Such newsletters suddenly take one to two seconds for each press of the DownArrow to read the next line and refresh the braille display. So the only way is to make UIA in Word the default, and if that's the case, also add some heuristics to suppress layout tables when reading.

### Additional context

This came from a discussion with acquaintances who claimed that NVDA was still not ready for most workplace scenarios due to poor outlook support. I did some comparative testing between JAWS and NVDA, and besides the performance improvements I got when I turned on UIA for Word Controls, found this to be a real blocker for most users.

### PR DESCRIPTION
**Pull request by michaelDCurran**
### Link to issue number:
Fixes #11430

### Summary of the issue:
Microsoft Outlook uses a Microsoft word document control to present emails when both reading and composing. NVDA is in the process of switching to using UI Automation to access Microsoft Word document controls due to major performance advantages, and also due to Microsoft no longer maintaining the Office object model.
However, in the UI automation implementation exposed by MS Word document controls, it is impossible to identify the difference between a data table and a layout table. Thus, NVDa reports all layout tables when reading, which is extremely annoying, as the majority of HTML emails use many nested layout tables.

### Description of how this pull request fixes the issue:
Tables are now classed as layout tables in MS Word document controls if the app is Outlook, and the table is read-only (Outlook is in message reading view).
Being marked as a layout table means that by default the table will not be reported, however the user can still turn on the Include Layout Tables option in NVDA's browse mode settings temporarily if there is a genuine data table they need to have reported.
 
### Testing strategy:
With use UIA in MS Word controls enabled in NVDA's advanced settings:
* Create a table in Microsoft Word and ensure that NVDA still reports it (number of rows / columns, cell's row and column number etc).
* Create a table in Microsoft Outlook composition view and ensure that NVDA still reports it (number of rows / columns, cell's row and column number etc).
* Open an HTML email in Outlook that has layout tables, and ensure that the table is not reported when moving into and through the table's content with the arrow keys.

### Known issues with pull request:
All table reporting will be suppressed when reading emails in Outlook, even data tables. However, waying up the extreme annoyance of having layout tables read verses not reporting some very rare data tables (where the user can manually still enforce reporting by turning on includeLayoutTables) I think this is okay.
Though it would be much better if Outlook exposed layout tables in an identifiable way, such as not supporting the UI Automation Table Pattern for layout tables.

### Change log entries:
Bug fixes:
* When reading emails in Outlook and NVDA is accessing the message with UI Automation, reporting of tables is now suppressed by default.

### Code Review Checklist:

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual testing.
- [x] API is compatible with existing addons.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers

### CODE REVIEW

**Comment by CyrilleB79** [comment]
It seems that for Word documents (legacy), it is checked that the table has borders.
Is this something that could additionally be checked with UIA?
I guess layout tables do not have borders. Even if many data table don't either, this would already be a progress for the few data tables with borders.
I understand that the HTML messages with many layout tables (such as newsletters) is the most frequent. Even if less used, data table are quite common in professionnal or technical messages.

**Comment by michaelDCurran** [comment]
No, UIA does not expose border information sadly.

File: source/NVDAObjects/UIA/wordDocument.py [hunk 1]
```diff
@@ -177,6 +177,15 @@ def _getControlFieldForUIAObject(self, obj, isEmbedded=False, startOfNode=False,
 				or field.pop('name', None)
 				or obj.name
 			)
+		# #11430: Read-only tables, such as in the Outlook message viewer
+		# should be treated as layout tables
+		# If they only contain 1 column or 1 row.
```
Comment (seanbudd) [inline source/NVDAObjects/UIA/wordDocument.py:182]: Where does the check for "1 column or 1 row" occur?

**Review by seanbudd — APPROVED** [review]

**Comment by michaelDCurran** [comment]
Oh, that was an out of date comment. Further discussion on #11430 chose 
against checking number of rows and columns.

**Pull request merged**